### ISSUE TITLE
Dynamic learning rate

### ISSUE THREAD

**Issue by shindavid** [shindavid/AlphaZeroArcade#224]
As discussed over Discord:

On each training generation, with probability `p`, do a "redundant" backprop. Instead of just a single pass of `M` minibatch gradient updates with learning rate `L`, try 3 independent passes:

1. Small learning rate: `L / 2`
2. Regular learning rate: `L`
3. Big learning rate: `2 * L`

If 1 or 3 wins, then try this again on the next generation. If you get the same winner on `K` consecutive generations, then do a permanent learning rate change (`L -> L / 2` if 1 won every time, and `L -> 2 * L` if 3 won every time). On the other hand, if you get a non-repeat result, go back to only trying with probability `p`.

Alternatively, instead of randomly doing this with probability `p`, do it every `N` generations, for a little more run-to-run consistency. Or, use a fixed random seed when sampling `p`.

The fact that this is not done in other settings tells me that this is probably not generally a good strategy. But in the AlphaZero setting, where self-play dominates the run-time (at least, it's supposed to for non-trivial games, it's not quite true in our toy games), perhaps there's a metric by which this strategy wins.

In a multi-machine setting, where you have one machine dedicated to training, it may sit idly much of the time, in which case you might as well just do this.
